src/ipc/ipc-common.c

```diff
@@ -178,6 +178,26 @@ static void schedule_ipc_worker(void)
 #endif
 }
 
+void ipc_msg_send_direct(struct ipc_msg *msg, void *data)
+{
+	struct ipc *ipc = ipc_get();
+	k_spinlock_key_t key;
+	int ret;
+
+	key = k_spin_lock(&ipc->lock);
+
+	/* copy mailbox data to message if not already copied */
+	if (msg->tx_size > 0 && msg->tx_size < SOF_IPC_MSG_MAX_SIZE &&
```

Comment:
actually, looking at it: shouldn't the second comparison be `<=`? "maximum size" usually is allowed too?